pulp_ansible/app/galaxy/views.py

```diff
@@ -234,12 +236,17 @@ def get(self, request, path, namespace, name, version):
             ContentArtifact, content__in=self._distro_content, relative_path=version.relative_path
         )
 
-        download_url = "{content_hostname}/{base_path}/{relative_path}".format(
-            content_hostname=settings.ANSIBLE_CONTENT_HOSTNAME,
-            base_path=self.kwargs["path"],
-            relative_path=version.relative_path,
-        )
-
-        data = GalaxyCollectionVersionSerializer(version).data
+        # Note: We're using ANSIBLE_API_HOSTNAME here instead of calling reverse with request=
+        # because using the request context to generate the full URL causes the download URL
+        # to be inaccessible when pulp is running behind a reverse proxy.
```

Comment:
That sounds a bit wrong to me. It is far to common to run a django app behind a reverse proxy.

Reply:
Yeah, I didn't actually write this code, as mentioned I copied it from elsewhere.  I don't have the django dev experience, and certainly not the pulp dev experience to really know.  I can test without the use of ANSIBLE_API_HOSTNAME in the container, since it does use nginx as a reverse proxy.

Reply:
btw, this is the commit that implemented the current code to generate the URL: https://github.com/pulp/pulp_ansible/commit/4e06679b268049856f9e6662eb800447a391c79c

It looks like there is some additional history behind not directly using `request`

Reply:
I talked to newswangerd about this, since I copied his code, and was told the code is acting like this because:

> The api service runs behind an nginx proxy, which isn’t configured to pass the original port to the service, you can’t redirect to the original url that the user hit

Reply:
Ok, I've rewritten it to use `reverse` and `request.build_absolute_uri`, the `version` kwarg caused issues here too.  This at least works for me in the container that uses an nginx proxy.